php/class-wp-seo-settings.php

```diff
@@ -1,797 +1,523 @@
+
+		// Single post types.
+		foreach ( $this->get_available_single_post_types() as $post_type ) {
+			$sanitize_as_text_field[] = "single_{$post_type}_title";
+			$sanitize_as_text_field[] = "single_{$post_type}_description";
+			$sanitize_as_text_field[] = "single_{$post_type}_keywords";
+		}
+
+		// Archived post types.
+		foreach ( $this->get_available_archive_post_types() as $post_type ) {
+			$sanitize_as_text_field[] = "archive_{$post_type}_title";
+			$sanitize_as_text_field[] = "archive_{$post_type}_description";
+			$sanitize_as_text_field[] = "archive_{$post_type}_keywords";
+		}
+
+		// Taxonomy archives.
+		foreach ( $this->get_available_taxonomies() as $taxonomy ) {
+			$sanitize_as_text_field[] = "archive_{$taxonomy}_title";
+			$sanitize_as_text_field[] = "archive_{$taxonomy}_description";
+			$sanitize_as_text_field[] = "archive_{$taxonomy}_keywords";
+		}
+
+		// Author and date archives.
+		foreach ( array( 'author', 'date' ) as $type ) {
+			$sanitize_as_text_field[] = "archive_{$type}_title";
+			$sanitize_as_text_field[] = "archive_{$type}_description";
+			$sanitize_as_text_field[] = "archive_{$type}_keywords";
+		}
+
+		// Search and 404 titles.
+		$sanitize_as_text_field[] = 'search_title';
+		$sanitize_as_text_field[] = '404_title';
+
+		// Finally, sanitize.
+		foreach ( $sanitize_as_text_field as $field ) {
+			if ( ! isset( $in[ $field ] ) || ! is_string( $in[ $field ] ) ) {
+				$in[ $field ] = '';
+			}
+
+			$out[ $field ] = sanitize_text_field( $in[ $field ] );
+		}
+
+		// Sanitize arbitrary tags, also as text fields.
+
+		if ( ! isset( $in['arbitrary_tags'] ) || ! is_array( $in['arbitrary_tags'] ) ) {
+			$in['arbitrary_tags'] = array();
+		}
+
+		$out['arbitrary_tags'] = array();
+
+		foreach ( $in['arbitrary_tags'] as $group ) {
+			// Skip if a field is missing.
+			if ( ! isset( $group['name'] ) || ! isset( $group['content'] ) ) {
+				continue;
+			}
+
+			$name = sanitize_text_field( $group['name'] );
+			$content = sanitize_text_field( $group['content'] );
+
+			// Skip if both fields are, or were sanitized into being, empty.
+			if ( empty( $name ) && empty( $content ) ) {
+				continue;
+			}
+
+			$out['arbitrary_tags'][] = array(
+				'name' => $name,
+				'content' => $content,
+			);
+		}
+
+		return $out;
+	}
 
 	/**
-	 * Post types that can be viewed individually and have per-entry meta values.
-	 *
-	 * @see  WP_SEO_Settings::setup().
+	 * Get the post types with per-post fields enabled.
 	 *
-	 * @var array Post type objects.
+	 * @return array Array of names of any enabled post types.
 	 */
-	private $single_post_types = array();
+	public function get_enabled_post_types() {
+		$post_types = $this->get_option( 'post_types', $this->get_available_single_post_types() );
+
+		// Remove post types that are saved but not enabled.
+		return array_filter( $post_types, function ( $name ) {
+			return in_array( $name, $this->get_available_single_post_types(), true );
```

Comment:
Travis is complaining about this use of `$this`. Is this `$this` legal inside an anonymous function? Even if so, it may be worth living with an extra variable to get around Travis and make the code less elegant but more readable.